packages/flutter_tools/test/general.shard/build_system/targets/icon_tree_shaker_test.dart

```diff
@@ -0,0 +1,428 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/dart.dart';
+import 'package:flutter_tools/src/build_system/targets/icon_tree_shaker.dart';
+import 'package:flutter_tools/src/devfs.dart';
+import 'package:meta/meta.dart';
+import 'package:mockito/mockito.dart';
+import 'package:platform/platform.dart';
+
+import '../../../src/common.dart';
+import '../../../src/context.dart';
+import '../../../src/mocks.dart' as mocks;
+
+final Platform _kNoAnsiPlatform =
+    FakePlatform.fromPlatform(const LocalPlatform())
+      ..stdoutSupportsAnsi = false;
+
+void main() {
+  BufferLogger logger;
+  MemoryFileSystem fs;
+  MockProcessManager mockProcessManager;
+  MockProcess fontSubsetProcess;
+  MockArtifacts mockArtifacts;
+  DevFSStringContent fontManifestContent;
+
+  const String dartPath = '/flutter/dart';
+  const String constFinderPath = '/flutter/const_finder.snapshot.dart';
+  const String fontSubsetPath = '/flutter/font-subset';
+
+  const String inputPath = '/input/fonts/MaterialIcons-Regular.ttf';
+  const String outputPath = '/output/fonts/MaterialIcons-Regular.ttf';
+  const String relativePath = 'fonts/MaterialIcons-Regular.ttf';
+
+  List<String> getConstFinderArgs(String appDillPath) => <String>[
+    dartPath,
+    constFinderPath,
+    '--kernel-file', appDillPath,
+    '--class-library-uri', 'package:flutter/src/widgets/icon_data.dart',
+    '--class-name', 'IconData',
+  ];
+
+  const List<String> fontSubsetArgs = <String>[
+    fontSubsetPath,
+    outputPath,
+    inputPath,
+  ];
+
+  void _addConstFinderInvocation(
+    String appDillPath, {
+    int exitCode = 0,
+    String stdout = '',
+    String stderr = '',
+  }) {
+    when(mockProcessManager.run(getConstFinderArgs(appDillPath))).thenAnswer((_) async {
+      return ProcessResult(0, exitCode, stdout, stderr);
+    });
+  }
+
+  void _resetFontSubsetInvocation({
+    int exitCode = 0,
+    String stdout = '',
+    String stderr = '',
+    @required List<String> stdinResults,
+  }) {
+    assert(stdinResults != null);
+    stdinResults.clear();
+    final IOSink sink = IOSink(StringStreamConsumer(stdinResults));
+    when(fontSubsetProcess.exitCode).thenAnswer((_) async => exitCode);
+    when(fontSubsetProcess.stdout).thenAnswer((_) => Stream<List<int>>.fromIterable(<List<int>>[utf8.encode(stdout)]));
+    when(fontSubsetProcess.stderr).thenAnswer((_) => Stream<List<int>>.fromIterable(<List<int>>[utf8.encode(stderr)]));
+    when(fontSubsetProcess.stdin).thenReturn(sink);
```

Comment:
An IOSink that throws on add() would test the error handling. See `CompleterIOSink` in mocks.dart.